src/assets/scripts/client/base/BaseModel.js

```diff
@@ -47,23 +49,17 @@ export default class BaseModel {
     }
 
     /**
-     * This will verify if the given argument is a string and will return a modified pre-value for the _id
+     * This will verify if the given argument is a string otherwise it will return 'Base Model'
      *
      * @for BaseModel
      * @method veriftyModelName
      * @param {string / Object}
      */
-    verifyModelName(modelName) {
-        if (typeof modelName === 'string') {
+    buildModelPrefix(modelName) {
+        if (_isString(modelName)) {
             return `${modelName}-`;
```

Comment:
This if logic should be reversed.  We should be exclusive, not inclusive, with our logic blocks.
```javascript
// instead of (inclusive)
if (_isString(modelName)) { 
  return `${modelName}-`;
}

// we should do (exclusive)
if (!_isString(modelName)) {
  return `${modelName}-`;
}
``` 
a method should eject (return early) as soon as we find data we don't want.

Also, since `modelName` is defaulted to `BaseModel` in the constructor, we shouldn't need to return that string again below.  We can just use `modelName-`.

So with that in mind, does the value we return change?  Doesn't look like it.  But we do need to do something in the case we receive bad data, and handle that appropriately.  In this case, we should use `console.error` if we receive a param that is not a string, but still return our `modelName-` because that kind of error shouldn't prevent the model from instantiating.